Kernel/Devices/Storage/USB/UAS/Structures.h

```diff
@@ -117,9 +119,19 @@ struct SenseIU {
     SCSI::StatusCode status;
     u8 reserved[7];
     BigEndian<u16> length; // FIXME: The spec does not actually state the endianness of this?
-    u8 sense_data[];
+    // Note: As a Sense IU is of technically of flexible size
+    //       we need to forcefully allocate some data here, to be able to put it
+    //       into an ErrorOr later
+    //       The maximum size of the sense data the device may send is controlled by the
+    //       MAXIMUM SENSE DATA LENGTH field in the Control extension mode page of the device.
+    //       In theory the maximum size of the sense data is 252 bytes
+    //       Meaning the maximum size of the IU is 252+16=268 bytes
+    //       Just to be safe we allocate 512 bytes here, as the spec also tells us that the
+    //       SenseIU is to not share a USB packet with any other IU, so we can just use
+    //       the maximum packet size instead, which is 512 bytes (USB3 allows up to 1024 bytes)
```

Comment:
a bit run-on-sentence-y

Reply:
Well this is moved from the big `union IU` comment, and only slightly adjusted

Reply:
Adjusted the comment again, is it better now?